crates/datasources/src/native/access.rs

```diff
@@ -56,12 +58,15 @@ pub struct NativeTableStorage {
     /// what this type is for.
     ///
     /// Arcs all the way down...
-    store: SharedObjectStore,
+    pub store: SharedObjectStore,
```

Comment:
q: should I provide a getter that performs a clone? (does that make sense from mem model)?

There's a comment here that describes that this object was created to be cloned:

https://github.com/GlareDB/glaredb/blob/a1787da8c70311782305ba7b5f6ce0997f370c58/crates/object_store_util/src/shared.rs#L28-L33

Reply:
I think the compiler does the thing that you would want it to do, so no need to get clever.